### ISSUE TITLE
JS function handling is very fragile

### ISSUE THREAD

**Issue by wichert** [i18next/i18next-scanner#50]
The parser is not very robust when it looks for translation functions. Changing `test/fixtures/app.js` to add some newlines breaks the tests:

```
i18next.t('key2');
i18next.t(
  'key1'
);
```

For my specific use case I would like to be able to handle this code:

```javascript
i18next.t(
  'errors.password-mismatch',
  {
    defaultValue: 'Your passwords did not match.'
  }
);
```

**Comment by wichert** [comment]
For what it's worth I tried to rewrite parseFuncFromString to use acorn or esprima to parse the entire file and then walk the AST. That did not work: both parsers would trip over things like object spread, or use of JSX syntax. I'm afraid I ran out of time to look at this further; hopefully this can be fixed by just changing the regexp pattern used (perhaps replace `.` with `[^]` in a few places?).

**Comment by cheton** [comment]
I added a `matchStrippedCharacters` to match white spaces and new lines.

```js
const matchFuncs = funcs
     .map(func => ('(?:' + func + ')'))
    .join('|')
    .replace(/\./g, '\\.');
const matchStrippedCharacters = '[\\r\\n\\s]*';
const pattern = '(?:(?:^\\s*)|[^a-zA-Z0-9_])(?:' + 
    matchFuncs + 
    ')\\((' + 
    matchStrippedCharacters + 
    '"(?:[^"\\\\]|\\\\(?:.|$))*"|' + 
    matchStrippedCharacters + 
    '\'(?:[^\'\\\\]|\\\\(?:.|$))*\')' + 
    matchStrippedCharacters + '[,)]';
```

Fixed in v2.0.4.

**Comment by wichert** [comment]
Thanks for the quick fix!
